ouroboros-network/src/Ouroboros/Network/Diffusion/P2P.hs

```diff
@@ -1001,6 +1011,7 @@ runM Interfaces
             makeConnectionHandler
               dtMuxTracer
               muxMode
+              (responderForkPolicy (fst $ random forkRng) numCapabilities)
```

Comment:
This behavior should be behind a feature flag and disabled by default in cardano-node.
Now it is always enabled.

Reply:
You're right. I added `daForkPolicy :: ForkPolicy peeraddr` to `ArgumentsExtra` so the library user can plug its own thing.